src/main/java/org/jenkinsci/remoting/engine/JnlpProtocol4Handler.java

```diff
@@ -301,7 +301,7 @@ public void onReceiveHeaders(Map<String, String> headers) throws ConnectionRefus
                     case INVALID:
                         LOGGER.log(Level.WARNING,
                                 "An attempt was made to connect as {0} from {1} with an invalid client certificate",
-                                new Object[]{clientName, event.getSocket().getRemoteSocketAddress()});
+                                new Object[]{clientName, event.getRemoteEndpointDescription()});
```

Comment:
The changes here are needed to adapt to the changes in `JnlpConnectionState` but those don't seem to be needed. These particular uses do make more sense, though.

Reply:
See above.